batch2/batch/batch.py

```diff
@@ -1,19 +1,1278 @@
 from aiohttp import web
+import cerberus
+import kubernetes as kube
+import requests
+import prometheus_client as pc
+from prometheus_async.aio import time as prom_async_time
+from prometheus_async.aio.web import server_stats
 
+from hailtop.utils import unzip, blocking_to_async
+from hailtop.auth import async_get_userinfo
 from hailtop.config import get_deploy_config
-from gear import configure_logging
+from gear import setup_aiohttp_session, \
+    rest_authenticated_users_only, web_authenticated_users_only, \
+    new_csrf_token, check_csrf_token
+from web_common import setup_aiohttp_jinja2, setup_common_static_routes, base_context
+
+import uvloop
+uvloop.install()
+
+from .blocking_to_async import blocking_to_async
+from .log_store import LogStore
+from .database import JobsBuilder
+from .datetime_json import JSON_ENCODER
+from .globals import states, complete_states, valid_state_transitions, tasks, get_db
+from .batch_configuration import KUBERNETES_TIMEOUT_IN_SECONDS, REFRESH_INTERVAL_IN_SECONDS, \
+    POD_VOLUME_SIZE, INSTANCE_ID, BATCH_IMAGE, BATCH_NAMESPACE
+from .driver import Driver
+from .k8s import K8s
+from .utils import abort, jsonify
+
+from . import schemas
+
+log = logging.getLogger('batch')
+
+REQUEST_TIME = pc.Summary('batch_request_latency_seconds', 'Batch request latency in seconds', ['endpoint', 'verb'])
+REQUEST_TIME_GET_JOB = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id/jobs/job_id', verb="GET")
+REQUEST_TIME_GET_JOB_LOG = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id/jobs/job_id/log', verb="GET")
+REQUEST_TIME_GET_POD_STATUS = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id/jobs/job_id/pod_status', verb="GET")
+REQUEST_TIME_GET_BATCHES = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches', verb="GET")
+REQUEST_TIME_POST_CREATE_JOBS = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id/jobs/create', verb="POST")
+REQUEST_TIME_POST_CREATE_BATCH = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/create', verb='POST')
+REQUEST_TIME_POST_GET_BATCH = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id', verb='GET')
+REQUEST_TIME_PATCH_CANCEL_BATCH = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id/cancel', verb="PATCH")
+REQUEST_TIME_PATCH_CLOSE_BATCH = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id/close', verb="PATCH")
+REQUEST_TIME_DELETE_BATCH = REQUEST_TIME.labels(endpoint='/api/v1alpha/batches/batch_id', verb="DELETE")
+REQUEST_TIME_GET_BATCH_UI = REQUEST_TIME.labels(endpoint='/batches/batch_id', verb='GET')
+REQUEST_TIME_POST_CANCEL_BATCH_UI = REQUEST_TIME.labels(endpoint='/batches/batch_id/cancel', verb='POST')
+REQUEST_TIME_GET_BATCHES_UI = REQUEST_TIME.labels(endpoint='/batches', verb='GET')
+REQUEST_TIME_GET_LOGS_UI = REQUEST_TIME.labels(endpoint='/batches/batch_id/jobs/job_id/log', verb="GET")
+REQUEST_TIME_GET_POD_STATUS_UI = REQUEST_TIME.labels(endpoint='/batches/batch_id/jobs/job_id/pod_status', verb="GET")
```

Comment:
I can dig up the resources about it again, but IIRC basically Prometheus stores a "time series" (which corresponds to a file / set of files) for every combination of labels. So if you have unbounded label sets, you end up using a lot more disk storage as a result of Prometheus's on disk representation choices.  It may also make querying slower (since more files to find read when you filter on a particular label).